SConstruct

```diff
@@ -79,7 +86,13 @@ def set_defaults(env):
                         '-fpic', '-D_GNU_SOURCE', '-DD_LOG_V2'])
     env.Append(CCFLAGS=['-O2', '-DDAOS_VERSION=\\"' + DAOS_VERSION + '\\"'])
     env.Append(CCFLAGS=['-DCMOCKA_FILTER_SUPPORTED=0'])
+    env.Append(CCFLAGS=['-D_FORTIFY_SOURCE=2'])
     env.AppendIfSupported(CCFLAGS=DESIRED_FLAGS)
+
+    compiler = env.subst("$COMPILER")
+    if compiler == "clang":
+        env.Append(CCFLAGS=['-Wno-unused-command-line-argument'])
```

Comment:
Why not just put this into DESIRED_FLAGS and it'll be used if the compiler supports it, that's what DESIRED_FLAGS is for.

Reply:
I figured since it was a clang specific flag it would be best to test for clang but we could do that as well. I'm somewhat reluctant to make further changes at the moment though since this PR has been through CI 4 times in the last week and It might actually pass this time. The failures have nothing to do with this PR but rather that CI is hit or miss at the moment.

Reply:
Since the PR has to go back through CI anyway I'll make this change.

Reply:
Fixed